src/Xamarin.Android.Build.Tasks/Tasks/ComputeHash.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Security.Cryptography;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks {
+	public class ComputeHash : Task {
+
+		List<ITaskItem> output = new List<ITaskItem> ();
+
+		[Required]
+		public ITaskItem [] Source { get; set; }
+
+		[Output]
+		public ITaskItem [] Output { get => output.ToArray (); }
+
+		public override bool Execute ()
+		{
+			Log.LogDebugTaskItems ("Source : ", Source);
+			using (var sha1 = SHA1.Create ()) {
+
+				foreach (var item in Source) {
+					output.Add (new TaskItem (item.ItemSpec, new Dictionary<string, string> () {
+						{ "Hash", HashItemSpec (sha1, item.ItemSpec) }
+					}));
+				}
+				Log.LogDebugTaskItems ("Output : ", Output);
+				return !Log.HasLoggedErrors;
+			}
+		}
+
+		string HashItemSpec (SHA1 sha1, string hashInput)
+		{
+			var hash = sha1.ComputeHash (Encoding.UTF8.GetBytes (hashInput));
+			var hashResult = new StringBuilder (hash.Length * 2);
+
+			foreach (byte b in hash) {
+				hashResult.Append (b.ToString ("x2"));
+			}
+			return hashResult.ToString ();
+		}
+	}
+}
```

Comment:
@jonathanpeppers the Hash function in msbuild hashes all the times provided into one big hash. 
The one we have returns a hash per item. Now we could call the msbuild one but we could have to use batching.. I'm not sure which is more efficient.. calling a task 50 times, or calling a task once and having it hash 50 items. I suspect the latter since we don't need to create and destroy the task each time.